collections/src/main/scala/strawman/collection/immutable/ImmutableArray.scala

```diff
@@ -285,5 +328,15 @@ object ImmutableArray extends StrictOptimizedSeqFactory[ImmutableArray] {
       case that: ofUnit => unsafeArray.length == that.unsafeArray.length
       case _ => super.equals(that)
     }
+    override def slice(from: Int, until: Int): ImmutableArray[Unit] = {
+      // cant use
+      // new ofUnit(util.Arrays.copyOfRange[Unit](array, from, until)) - Unit is special and doesnt compile
+      // cant use util.Arrays.copyOfRange[Unit](repr, from, until) - Unit is special and doesnt compile
+      val lo = scala.math.max(from, 0)
+      val hi = scala.math.min(until, length)
+      val slicedLenght = hi - lo
+      val res = new Array[Unit](slicedLenght)
```

Comment:
Need to make sure the length is non-negative.

Reply:
thanks, fixed